feat: added support for ancestry.com and docker compose

mattboston opened the pull request:
This PR introduces comprehensive support for Ancestry.com DNA data files and adds Docker containerization for easier deployment and development.

Key Features Added

- Ancestry.com DNA File Support: Enhanced CSV parser to handle Ancestry.com's specific file format and data structure
- Docker Containerization: Complete Docker setup with Dockerfile and docker-compose.yml for containerized deployment
- Improved File Processing: Enhanced file processor to handle multiple DNA testing service formats
- UI Enhancements: Updated user interface to better support Ancestry.com data visualization
- Documentation Updates: Comprehensive README updates with Docker setup instructions

mattboston committed:
feat: added support for ancestry.com and docker compose

mattboston commented:
Just wanted to check in and see if there are any changes you want me to make

hartmark commented:
I'm sorry I haven't gotten back to you earlier. I'm on vacation so I'm not checking my mail so often and don't have so much extra time to fiddle.

I have checked your changes and it looks good, but it seems to have broken the parsing of my MyHeritage files.

I get this error now in console:
Error processing MyHeritage_raw_dna_data_markus.zip: Could not determine the DNA file format. Please ensure this is a valid MyHeritage or Ancestry.com DNA file.

But the gui doesn't show any errors to the user.

this is how the csv-file looks like in the start if you want to dig in and see what gets wrong.

```
##fileformat=MyHeritage
##format=MHv1.0
##chip=GSA
##timestamp=2025-04-20 17:12:19 UTC
##reference=build37
#
# MyHeritage DNA raw data.
# For each SNP, we provide the identifier, chromosome number, base pair position and genotype.
# The genotype is reported on the forward (+) strand with respect to the human reference build 37.
# THIS INFORMATION IS FOR YOUR PERSONAL USE AND IS INTENDED FOR GENEALOGICAL RESEARCH
# ONLY. IT IS NOT INTENDED FOR MEDICAL OR HEALTH PURPOSES. PLEASE BE AWARE THAT THE
# DOWNLOADED DATA WILL NO LONGER BE PROTECTED BY OUR SECURITY MEASURES.
RSID,CHROMOSOME,POSITION,RESULT
"rs547237130","1","72526","AA"
```

hartmark commented:
also the port 8000 was in use already on my machine, I think it would be better to use some more random port number than round numbers.

or even better use a default port and allow ovverides in .env

`% git diff`
```
diff --git a/docker-compose.yml b/docker-compose.yml
index 34eecd6..3c9e4c9 100644
--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -5,7 +5,7 @@ services:
     image: node:18-alpine
     container_name: dnahacker
     ports:
-      - "8000:8000"
+      - "${APP_PORT:-8000}:8000"
     volumes:
       - .:/app
     working_dir: /app
```

`% cat .env `
`APP_PORT=8001`

mattboston committed:
fix: fixed myheritage processing, added APP_PORT and .gitignore

mattboston commented:
Sounds good.

Thanks for the example MyHeritage file. I was able to fix that processing.

Also added a .gitignore to make sure we don't upload our personal data, and added the APP_PORT var.

hartmark committed:
remove obsolete version tag

hartmark commented:
I have merged your PR now with just a small fix. Thanks for your contribution!
